In [42]:
import pandas as pd

### Scenario 1

In [43]:
df = pd.DataFrame()
df['ages'] = [14,12,11,10,8,6,8]

In [44]:
df.mean()

ages    9.857143
dtype: float64

In [45]:
df.mode()

ages
0     8

In [46]:
df.median()

ages    10.0
dtype: float64

In [48]:
df.var(ddof=1)

ages    7.47619
dtype: float64

In [47]:
df.std(ddof=1)

ages    2.734262
dtype: float64

In [49]:
df.std(ddof=1)/(df['ages'].size**0.5)

ages    1.033454
dtype: float64

### Scenario 2

### "Median" is a better estimate of central tendency, because more than half of the population is older than mode and mean.

### "Std Deviation" of 2.734262 covers 5/7 values i.e fall in range of (mean +- std_deviation) i.e  71.4% of the population is actually covered. Since we are not concerned with sampling the population, Std Error may not be a relevant measure of variance.


### Scenario 3

In [50]:
df1 = pd.DataFrame()
df1['ages'] = [14,12,11,10,8,7,8]

In [51]:
df1.mean()

ages    10.0
dtype: float64

In [52]:
df1.mode()

ages
0     8

In [53]:
df1.median()

ages    10.0
dtype: float64

In [63]:
df1.var(ddof=1)

ages    6.333333
dtype: float64

In [64]:
df1.std(ddof=1)

ages    2.516611
dtype: float64

In [65]:
df1.std(ddof=1)/(df1['ages'].size**0.5)

ages    0.95119
dtype: float64

### "Median" and "Mean" is a better estimate of central tendency, because more than half of the population is older than mode.

### Scenario 4

In [59]:
df2 = pd.DataFrame()
df2['ages'] = [14,12,11,10,8,7,1]

In [60]:
df2.mean()

ages    9.0
dtype: float64

In [61]:
df2.mode()

Empty DataFrame
Columns: [ages]
Index: []

In [62]:
df2.median()

ages    10.0
dtype: float64

In [66]:
df2.var(ddof=1)

ages    18.0
dtype: float64

In [67]:
df2.std(ddof=1)

ages    4.242641
dtype: float64

In [68]:
df2.std(ddof=1)/(df2['ages'].size**0.5)

ages    1.603567
dtype: float64

### "Median" is a better estimate of central tendency, because more than half of the population is older than mean.